plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2087,26 +2090,37 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         }
 
         Optional<ConnectorTablePartitioning> tablePartitioning = Optional.empty();
-        if (isBucketExecutionEnabled(session) && hiveTable.getBucketHandle().isPresent()) {
-            tablePartitioning = hiveTable.getBucketHandle().map(bucketing -> new ConnectorTablePartitioning(
-                    new HivePartitioningHandle(
-                            bucketing.getBucketingVersion(),
-                            bucketing.getReadBucketCount(),
-                            bucketing.getColumns().stream()
-                                    .map(HiveColumnHandle::getHiveType)
-                                    .collect(toImmutableList()),
-                            OptionalInt.empty()),
-                    bucketing.getColumns().stream()
-                            .map(ColumnHandle.class::cast)
-                            .collect(toImmutableList())));
+        List<LocalProperty<ColumnHandle>> sortingProperties = ImmutableList.of();
+        if (hiveTable.getBucketHandle().isPresent()) {
+            if (isOptimizeSortedFileReads(session) && !hiveTable.getBucketHandle().get().getSortedBy().isEmpty()) {
+                Map<String, ColumnHandle> columnHandles = getColumnHandles(session, table);
+                sortingProperties = hiveTable.getBucketHandle().get().getSortedBy().stream()
```

Comment:
Maybe add a comment why this is (needs) be conditional on a session property. We have discussed this, but let;s keep this knowledge with the code.

Reply:
Added a comment here